### PR TITLE
[docker] bogus pid

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

This PR attempts to address the case where we either have a bogus container pid reported, or no pid was collected, in which case the container must be skipped. 

### Motivation

Issue has been affecting some customers.

### Testing Guidelines

Further testing must be conducted to reproduce this reliably, I have been unable to so far. But then again I haven't had the time to dedicate a solid half-day to reproducing.

An overview on [testing](https://github.com/DataDog/dd-agent/blob/master/tests/README.md)
is available in our contribution guidelines.

### CODE REVIEW

File: checks.d/docker_daemon.py [hunk 1]
```diff
@@ -575,11 +575,14 @@ def _report_performance_metrics(self, containers_by_id):
...
-                continue
-            self._report_net_metrics(container, tags)
+            try:
+                self._report_cgroup_metrics(container, tags)
+                if "_proc_root" not in container:
+                    containers_without_proc_root.append(DockerUtil.container_name_extractor(container)[0])
+                    continue
+                self._report_net_metrics(container, tags)
+            except BogusPIDException:
+                self.log.exception('Unable to report cgroup metrics.')
```
Comment (hkaj) [inline checks.d/docker_daemon.py]: This is still going to fill the logs with tracebacks. Maybe `log.error` is enough? With the container id or name for reference.

File: checks.d/docker_daemon.py [hunk 2]
```diff
@@ -592,6 +595,9 @@ def _report_performance_metrics(self, containers_by_id):
 
     def _report_cgroup_metrics(self, container, tags):
         cgroup_stat_file_failures = 0
+        if not container.get('_pid'):
+            raise BogusPIDException('Cannot report on bogus pid(0)')
```
Comment (hkaj) [inline checks.d/docker_daemon.py]: does that cover `0`-ed PIDs?

Comment (truthbk) [inline checks.d/docker_daemon.py]: it does

File: checks.d/docker_daemon.py (hunk 1 above)
Comment (truthbk) [inline checks.d/docker_daemon.py]: sounds good, I'll update that.

**Comment by truthbk** [comment]
closing this one, opened here:

**Pull request closed without merging**